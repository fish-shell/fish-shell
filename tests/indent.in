echo Test1
echo -n '
begin
echo hi


end | cat | cat | begin ; echo hi ; end | begin ; begin ; echo hi ; end ; end arg
' | ../test/root/bin/fish_indent

echo \nTest2
echo -n '
switch aloha

   case alpha
      echo sup

   case beta gamma
      echo hi

end
' | ../test/root/bin/fish_indent

echo \nTest3
echo -n '
function hello_world

  'begin'
     echo hi
                     end | cat

   echo sup; echo sup
   echo hello;

   echo hello
                  end
' | ../test/root/bin/fish_indent

echo \nTest4
echo -n '
echo alpha         #comment1
#comment2

#comment3
for i in abc #comment1
   #comment2
  echo hi
end

switch foo #abc
   # bar
   case bar
       echo baz\
qqq
   case "*"
       echo sup
end' | ../test/root/bin/fish_indent

echo \nTest5
echo -n '
if true
else if false
    echo alpha
switch beta
   case gamma
       echo delta
end
end
' | ../test/root/bin/fish_indent -i

echo \nTest6
# Test errors
echo -n '
begin
echo hi
else
echo bye
end; echo alpha "
' | ../test/root/bin/fish_indent

echo \nTest7
# issue 1665
echo -n '
if begin ; false; end; echo hi ; end
while begin ; false; end; echo hi ; end
' | ../test/root/bin/fish_indent

echo \nTest redir formatting
# issue 2899
echo -n '
echo < stdin >>appended yes 2>&1 no > stdout maybe 2>&    4 | cat 2>| cat
' | ../test/root/bin/fish_indent

echo \nTest normalization of keywords
# issue 2921
echo -n '
i\
f true
    echo yes
en\
d

"whil\
e" true
    "builtin" yes
en"d"
' | ../test/root/bin/fish_indent
