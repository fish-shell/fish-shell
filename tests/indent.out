Test1

begin
    echo hi


end | cat | cat | begin
    echo hi
end | begin
    begin
        echo hi
    end
end arg

Test2

switch aloha

    case alpha
        echo sup

    case beta gamma
        echo hi

end

Test3

function hello_world

    begin
        echo hi
    end | cat

    echo sup
    echo sup
    echo hello


    echo hello
end

Test4

echo alpha #comment1
#comment2

#comment3
for i in abc #comment1
    #comment2
    echo hi
end

switch foo #abc
    # bar
    case bar
        echo baz\
qqq
    case "*"
        echo sup
end
Test5

if true
else if false
echo alpha
switch beta
case gamma
echo delta
end
end

Test6

begin
    echo hi
else
echo bye
end
echo alpha "

Test7

if begin
        false
    end
    echo hi
end
while begin
        false
    end
    echo hi
end

Test redir formatting

echo <stdin >>appended yes 2>&1 no >stdout maybe 2>&4 | cat 2>| cat

Test normalization of keywords

if true
    echo yes
end

while true
    builtin yes
end
